2015-07-31/dario_ruby_exceptions.rb

```diff
@@ -0,0 +1,102 @@
+# ERROR HANDLING AND EXCEPTIONS
+#
+# What is an exception?
+# -> unacceptable behaviour in a program
+#
+# Exception:
+#  -> is an object (a special kind)
+#  -> instance of the class Exception, or a class of one of its descendants
+#
+# Raising (an exception):
+#  -> means stopping normal execution of the program, and either dealing with an exception or exiting
+#     the entire program
+#  -> we deal with exceptions by providing a program with a |rescue| clause
+#
+# -> common exceptions:
+#  * RuntimeError - default error raise using the method |raise|
+#  * NoMethodError - object can't resolve a name of the message it receives to a method.
+#  * NameError - interpreter comes across an identifier it can't resolve. It can be variable name,
+#     method name.
+#  * IOError - error caused by trying to read a closed file, write to a read-only file etc.
+#  * TypeError - method receives an argument with the wrong type.
+#  * ArgumentError - method receives incorrect number of arugments.
+# -> they are all descendants of the class Exception
+#
+# Rescuing with a *rescue block*:
+#  -> begin > rescue > end
+
+# 9 - creating custom error classes
+#class HeroError < Exception; end
+HeroError = Class.new(Exception)
```

Comment:
👍

On Sat, Aug 1, 2015 at 2:47 AM, Zoran Majstorovic <notifications@github.com>
wrote:

> In 2015-07-31/dario_ruby_exceptions.rb
> <https://github.com/rubyzg/ruby-burgers/pull/2#discussion_r36028165>:
>
> > +# -> common exceptions:
> > +#  * RuntimeError - default error raise using the method |raise|
> > +#  * NoMethodError - object can't resolve a name of the message it receives to a method.
> > +#  * NameError - interpreter comes across an identifier it can't resolve. It can be variable name,
> > +#     method name.
> > +#  * IOError - error caused by trying to read a closed file, write to a read-only file etc.
> > +#  * TypeError - method receives an argument with the wrong type.
> > +#  * ArgumentError - method receives incorrect number of arugments.
> > +# -> they are all descendants of the class Exception
> > +#
> > +# Rescuing with a *rescue block*:
> > +#  -> begin > rescue > end
> > +
> > +# 9 - creating custom error classes
> > +#class HeroError < Exception; end
> > +HeroError = Class.new(Exception)
>
> better as HeroError = Class.new(StandardError) [image: :sparkles:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rubyzg/ruby-burgers/pull/2/files#r36028165>.
>